pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -2159,7 +2159,7 @@ public void refreshSegment(String tableNameWithType, SegmentMetadata segmentMeta
     sendSegmentRefreshMessage(tableNameWithType, segmentName, true, true);
   }
 
-  public int reloadAllSegments(String tableNameWithType, boolean forceDownload) {
+  public Pair<Integer, String> reloadAllSegments(String tableNameWithType, boolean forceDownload) {
```

Comment:
Let's also track task id for single segment reload

Reply:
Ack. I'm planning to add it post getting a general agreement on the approach with all segments APIs first. Is that okay?

Reply:
Have added these changes as well.